src/_zkapauthorizer/tests/test_client_resource.py

```diff
@@ -719,203 +751,208 @@ def create(store, state):
-            api_auth_token,
-        )
-        reason = "some interesting information"
-        fail_downloader = make_fail_downloader(Exception(reason))
-        get_fail_downloader = lambda cap: fail_downloader
-        root = root_from_config(config, aware_now, get_fail_downloader)
-        agent = RequestTraversalAgent(root)
+        # hook into the protocol's error-handling methods
+        messages = []
+        closes = []
+        proto.sendClose = lambda *args, **kw: closes.append((args, kw))
+        proto.sendMessage = lambda *args, **kw: messages.append((args, kw))
 
-        # Kick off the recovery attempt.
-        self.assertThat(
-            authorized_request(
-                api_auth_token,
-                agent,
-                b"POST",
-                b"http://127.0.0.1/recover",
-                headers=self.GOOD_REQUEST_HEADER,
-                data=BytesIO(self.GOOD_REQUEST_BODY),
-            ),
-            succeeded(Always()),
-        )
+        # run test by sending the initial message
+        proto.onMessage(message, False)
 
-        # Now inspect the recoverer's reported state.
-        requesting = authorized_request(
-            api_auth_token,
-            agent,
-            b"GET",
-            b"http://127.0.0.1/recover",
-        )
+        # all errors should result in a close message
         self.assertThat(
-            requesting,
-            succeeded(
-                matches_response(
-                    code_matcher=Equals(OK),
-                    headers_matcher=application_json(),
-                    body_matcher=matches_json(
+            closes,
+            MatchesListwise(
+                [
+                    AfterPreprocessing(
+                        lambda args_kwargs: args_kwargs[1],
                         Equals(
                             {
-                                "stage": "download_failed",
-                                "failure-reason": reason,
+                                "code": 4000,
+                                "reason": "Failed to parse recovery request",
                             }
                         ),
                     ),
+                ]
+            ),
+        )
+        return messages
+
+    @given(
+        tahoe_configs(),
+        api_auth_tokens(),
+    )
+    def test_status(self, get_config, api_auth_token) -> None:
+        """
+        A first websocket that initiates a recovery sees the same messages
+        as a second client (that uses the same dircap).
+        """
+        downloads = []
+        downloading_d: Deferred[None] = Deferred()
+
+        def get_success_downloader(cap):
+            async def do_download(set_state):
+                await downloading_d
+                downloads.append(set_state)
+                return (
+                    # this data is CBOR for {"version": 1, "statements": []}
+                    lambda: BytesIO(b"\xa2gversion\x01jstatements\x80"),
```

Comment:
This could be `statements_to_snapshot([])`.

Reply:
Did think of that ... would that be better, or better to have "concrete data that doesn't change"...?